lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -237,8 +237,16 @@ tokenize([$~, S, H | _] = Original, Line, Column, _Scope, Tokens) when ?is_upcas
 
 tokenize([$?, $\\, H | T], Line, Column, Scope, Tokens) ->
   Char = elixir_interpolation:unescape_map(H),
+  NewScope = if
+    H =:= Char andalso H =/= $\\ ->
```

Comment:
In your original comment you suggested: `(H =:= Char) and (H =/= $\\)`

1) Is there a reason to prefer `and` to `andalso` here?
2) Would you prefer the parentheses around each clause?

Reply:
Precedence in Erlang is not clear to me, so usually err on the side of caution. But in this case, we are inside an if, which allows guards, so I would do:

```suggestion
    H =:= Char, H =/= $\\ ->
```

Reply:
I didn't even know that was an option! Pushed.